src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,90 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Tabs.scss';
+
+class Tabs extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        const { activeTabId } = props;
+
+        this.state = {
+            activeTabId,
+            derivedProp: activeTabId // eslint-disable-line react/no-unused-state
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.activeTabId === state.derivedProp) {
+            return null;
+        }
+        return {
+            activeTabId: props.activeTabId,
+            derivedProp: props.activeTabId
+        };
+    }
+
+    render() {
+        const { activeTabId: idFromProps, children, onChange, ...rest } = this.props;
+        const { activeTabId } = this.state;
+
+        return (
+            <div {...rest} {...this.block()}>
+                <div {...this.elem('container')}>
+                    {React.Children.map(children, tab => {
+                        const { href, id } = tab.props;
+                        if (!id) {
+                            return null;
+                        }
+                        const active = activeTabId === id;
+                        return (
+                            <a
+                                href={href || '#'}
+                                {...this.elem(active ? 'tabActive' : 'tab')}
+                                key={id}
+                                onClick={e => {
```

Comment:
Can event handler  be extracted to a handleTabSwitch() method for better readability?

Reply:
Yes, makes sense to me